### PR TITLE
ci: use checkout action

### PR DESCRIPTION
**Pull request by 0x501D**
Manual git clone works bad.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -15,55 +15,45 @@ permissions:
...
     runs-on: [ self-hosted, linux-do ]
-    container: grafin1992/nemu-ci:v1.2
+    container: voidso/nemu-ci:v1.0
     needs: clone
 
     steps:
       - name: Configure nEMU build
-        working-directory: ./nemu
         run: |
-          mkdir build && cd build
```
Comment (grafin) [inline .github/workflows/linux.yml:43]: Decided to build inplace? This also tested, that out-of-tree build works fine, maybe leave it as is?

File: .github/workflows/linux_musl.yml [hunk 2]
```diff
@@ -9,41 +9,32 @@ on:
...
     runs-on: [ self-hosted, linux-do ]
-    container: grafin1992/nemu-ci:v1.2
+    container: voidso/nemu-musl-ci:v1.1
     needs: clone
 
     steps:
       - name: Configure nEMU build
-        working-directory: ./nemu_musl
         run: |
-          mkdir build && cd build
```
Comment (grafin) [inline .github/workflows/linux_musl.yml:37]: Same as above.

File: ci/images/alpine/Dockerfile [hunk 3]
```diff
@@ -2,4 +2,5 @@ FROM alpine
 
 RUN apk add --no-cache bash git cmake make gcc musl-dev pkgconfig \
     openssl-dev ncurses-dev libusb-dev eudev-dev json-c-dev libxml2-dev \
-    sqlite-dev libarchive-dev gettext-dev graphviz-dev dbus-dev
```
Comment (grafin) [inline ci/images/alpine/Dockerfile:5]: I was doing my best not to include nodejs in our CI... Dark times have come.

**Review by grafin — COMMENTED** [review]
Some minor comments. LGTM even if you leave it as is.

File: ci/images/alpine/Dockerfile (hunk 3 above)
Comment (0x501D) [reply ci/images/alpine/Dockerfile:5]: unfortunately the node is needed for gh actions

File: .github/workflows/linux_musl.yml (hunk 2 above)
Comment (0x501D) [reply .github/workflows/linux_musl.yml:37]: Added step

File: .github/workflows/linux.yml (hunk 1 above)
Comment (0x501D) [reply .github/workflows/linux.yml:43]: Added step

**Pull request merged**